tests/testutils.py

```diff
@@ -407,6 +409,62 @@ def skip_if_windows(cls):
     return decorator(cls)
 
 
+def crdb_version(conn, __crdb_version=[]):
+    """
+    Return the CockroachDB version if that's the db being tested, else None.
+
+    Return the number as an integer similar to PQserverVersion: return
+    v20.1.3 as 200103.
+
+    Assume all the connections are on the same db: return a cached result on
+    following calls.
+
+    """
+    if __crdb_version:
+        return __crdb_version[0]
+
+    sver = conn.info.parameter_status("crdb_version")
+    if sver is None:
+        __crdb_version.append(None)
+    else:
+        m = re.search(r"\bv(\d+)\.(\d+)\.(\d+)", sver)
+        if not m:
+            raise ValueError(
+                "can't parse CockroachDB version from %s" % sver)
+
+        ver = int(m.group(1)) * 10000 + int(m.group(2)) * 100 + int(m.group(3))
+        __crdb_version.append(ver)
+
+    return __crdb_version[0]
+
+
+def skip_if_crdb(reason, conn=None):
```

Comment:
I wonder if the decorator could also accept an optional issue number that refers to the CockroachDB repo. For most of the failing tests, we have an issue number that tracks why the test fails. See: https://github.com/cockroachdb/cockroach/blob/28c0337df4b695b6eba09639ca312a92348b9294/pkg/cmd/roachtest/psycopg_blocklist.go#L32

Reply:
Mmm sounds good, I can add it and use the go file to populate them

Reply:
Added. My list of bugs is slightly different: many were marked failed e.g. because temp table weren't supported: I think that feature was added later. I've compiled a hopefully more up-to-date list.
```
     28 not supported on CockroachDB: 2-phase commit (https://github.com/cockroachdb/cockroach/issues/22329)
      4 not supported on CockroachDB: backend pid (https://github.com/cockroachdb/cockroach/issues/35897)
      3 not supported on CockroachDB: cancel (https://github.com/cockroachdb/cockroach/issues/41335)
      1 not supported on CockroachDB: cast adds tz (https://github.com/cockroachdb/cockroach/issues/51692)
      3 not supported on CockroachDB: cidr (https://github.com/cockroachdb/cockroach/issues/18846)
     13 not supported on CockroachDB: composite (https://github.com/cockroachdb/cockroach/issues/27792)
      3 not supported on CockroachDB: connect any db
     19 not supported on CockroachDB: copy (https://github.com/cockroachdb/cockroach/issues/41608)
      1 not supported on CockroachDB: deferrable (https://github.com/cockroachdb/cockroach/issues/48307)
      1 not supported on CockroachDB: diagnostic
      3 not supported on CockroachDB: encoding (https://github.com/cockroachdb/cockroach/issues/35882)
      2 not supported on CockroachDB: flush on write flakey
      9 not supported on CockroachDB: hstore (https://github.com/cockroachdb/cockroach/issues/41284)
      1 not supported on CockroachDB: infinity date (https://github.com/cockroachdb/cockroach/issues/41564)
      1 not supported on CockroachDB: interval style (https://github.com/cockroachdb/cockroach/issues/35807)
     25 not supported on CockroachDB: isolation level
     16 not supported on CockroachDB: json (https://github.com/cockroachdb/cockroach/issues/)
     45 not supported on CockroachDB: large objects (https://github.com/cockroachdb/cockroach/issues/243)
     40 not supported on CockroachDB: named cursor (https://github.com/cockroachdb/cockroach/issues/41412)
      5 not supported on CockroachDB: nested array (https://github.com/cockroachdb/cockroach/issues/32552)
      8 not supported on CockroachDB: notice
     14 not supported on CockroachDB: notify (https://github.com/cockroachdb/cockroach/issues/41522)
     14 not supported on CockroachDB: range (https://github.com/cockroachdb/cockroach/issues/41282)
      1 not supported on CockroachDB: stored procedure (https://github.com/cockroachdb/cockroach/issues/1751)
      1 not supported on CockroachDB: table oid
```